framework/rpc/internal/server.cpp

```diff
@@ -0,0 +1,68 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2020 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+#include "server.h"
+#include "log.h"
+#include <algorithm>
+using namespace mu::rpc;
+
+Server::Server()
+    : m_messages(), m_targets()
+{
+}
+
+void Server::registerTarget(target_id id, std::shared_ptr<ITarget> target)
+{
+    m_targets[id] = target;
+}
+
+void Server::addCall(Message message)
+{
+    m_messages.push(message);
```

Comment:
not thread-safe (accessed from different threads)

Reply:
@igorkorsukov mutex for messages added. Thanks